mlflow/pyfunc/model.py

```diff
@@ -73,16 +73,59 @@ def _log_warning_if_params_not_in_predict_signature(logger, params):
         )
 
 
-class PythonModel:
+class _PythonModelMetaclass(ABCMeta):
+    def __call__(cls, *args, **kwargs):
```

Comment:
From GPT4—we override the `__call__` function in a metaclass, which gets called when subclasses get instantiated. It's probably unlikely that users will override this, and we inspect `__init__` from in here. What do yall think?

cc @harupy @BenWilson2

Reply:
Can we define `__new__` in `PythonModel` and fire the warning there?

Reply:
I like this idea :) (`__new__`)